Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/StitchByBackground.py

```diff
@@ -0,0 +1,150 @@
+            doc="Values in Q space where the stitch should take place. i.e. Overlaps in the data.",
+        )
+        self.declareProperty(
+            "OverlapWidth",
+            0.05,
+            validator=FloatBoundedValidator(lower=0),
+            doc="The extent to which the fit limits extend from the points given in StitchPoints.",
+        )
+        self.declareProperty("CropLowerBound", 0.0, doc="The XMin in Q-space to use when cropping the output workspace.")
+        self.declareProperty("CropUpperBound", 0.0, doc="The XMax in Q-space to use when cropping the output workspace.")
+
+    def PyExec(self):
+        ws_name_list = self.getProperty("InputWorkspaces").value
+        ws_list = [AnalysisDataService.retrieve(ws_name) for ws_name in ws_name_list]
+
+        stitch_locations = self.getProperty("StitchPoints").value
+
+        offsets = self.create_offsets(ws_list, stitch_locations)
+
+        subtracted_ws, x, y, e = self.subtract_background(ws_list, offsets, stitch_locations)
+
+        out_ws_name = self.getProperty("OutputWorkspace").value
+        stitched_ws = CreateWorkspace(
+            x,
+            y,
+            e,
+            OutputWorkspace=out_ws_name,
+            UnitX=subtracted_ws.getAxis(0).getUnit().unitID(),
+            Distribution=subtracted_ws.isDistribution(),
+            ParentWorkspace=subtracted_ws,
+            EnableLogging=False,
+        )
+
+        stitched_ws = self.crop_output(stitched_ws)
+
+        self.setProperty("OutputWorkspace", stitched_ws)
+
+    def create_offsets(self, workspaces, stitch_points):
+        overlap_width = self.getProperty("OverlapWidth").value
+        offsets = np.zeros(len(workspaces))
+        for stitch_index, stitch_point in enumerate(stitch_points):
+            backgrounds = []
+            # Get backgrounds for each side of the stitch.
+            for side in range(2):
+                ws = workspaces[stitch_index + side]
+                i_bin_of_stitch = ws.yIndexOfX(stitch_point)
+                background_func = f"name=LinearBackground, A0={ws.readY(0)[i_bin_of_stitch]}, A1={0};"
+                fit_result = Fit(
+                    Function=background_func, InputWorkspace=ws, StartX=stitch_point - overlap_width, EndX=stitch_point + overlap_width
+                )
+                backgrounds.append(fit_result.Function(stitch_point))
+            offsets[stitch_index] = backgrounds[1] - backgrounds[0]
+        return offsets
+
+    def subtract_background(self, ws_list, offsets, stitch_locations):
+        x = []
+        y = []
+        e = []
+        subtracted_ws = None
+        x_min = self.getProperty("CropLowerBound").value
+        x_max = self.getProperty("CropUpperBound").value
+        for ws_index, ws in enumerate(ws_list):
+            background_ws = CreateSingleValuedWorkspace(offsets[ws_index:].sum(), ErrorValue=0)
+            subtracted_ws = Plus(ws, background_ws, OutputWorkspace=f"bank{ws_index + 1}_offset")
```

Comment:
In my script I wanted to keep these offset workspaces for debug reasons, but here they probably don't need to be stored in the ADS